src/workloads/execution/Densify.yml

```diff
@@ -0,0 +1,749 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload test tests the performance of the $densify stage. $densify can operate on both dates
+  and numbers.
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000
+    BatchSize: &batchSize 30000 # Documents will be roughly 5 bytes in size, and we won't generate more than 30k documents, which is less than 1mb.
+    Document:
+      partitionKey: {^RandomInt: {min: 1, max: 100}}
+      # Test a subset of units that have different characteristics.
+      number: {^RandomInt: {min: 0, max: 20000}} # Seperate codepath for numeric DensifyValues.
+      milliseconds: {^RandomDate: {min: "2021-01-01T00:00:00.000Z", max: "2021-01-01T00:00:20.000Z"}} # Most basic date unit.
+      hours: {^RandomDate: {min: "2021-01-01", max: "2023-01-01"}} # Date unit with constant number of milliseconds.
+      months: {^RandomDate: {min: "2021-01-01", max: "4000-01-01"}} # Date unit with non-constant number of milliseconds.
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+
+# $densify stages always add a $sort on the field that is being densified, so this phase adds 
+# indexes on the date/number fields so that we aren't performing an in-memory sort.
+- Name: CreateIndex
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        createIndexes: Collection0
+        indexes: 
+        - key: {number: 1}
+          name: number
+        - key: {milliseconds: 1}
+          name: milliseconds
+        - key: {hours: 1}
+          name: hours
+        - key: {months: 1}
+          name: months
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+  - *Nop
+
+- Name: Densify
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 10
+    Database: *db
+    Operations:
+
+    # Metrics are grouped by the step unit they operate with (numeric, millisecond, hour, month).
+    # Each group has a metric for each possible codepath (Partition/no partition, explicit/full/partitioned bounds).
+
+    # Numeric metrics. 
+    - OperationMetricsName: NumericFullDensifySmallStep
```

Comment:
Nitpicky comment (but perhaps an annoyingly manual one) - for ease of comprehension could you interleave the small/large step operations with the same densify type? Aka NumericFullDensifySmallStep, and then right after, NumericFullDensifyLargeStep. This way readers can easily see the difference between these two queries, instead of having to look through each $densify combination for a small step and then doing it again for a large step.